### PR TITLE
optimized ResolveMojo and bug fixed

### PR DESCRIPTION
**Pull request by yegor256**

### CODE REVIEW

**Comment by yegor256** [comment]
@volodya-lombrozo please review, it's a bit urgent

File: eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java [hunk 1]
```diff
@@ -213,95 +210,23 @@ private Collection<Dependency> deps() throws IOException {
...
-                dep.getGroupId(), dep.getArtifactId(), dep.getVersion()
-            );
-        } else {
-            ret = String.format(
-                "%s:%s:%s:%s",
-                dep.getGroupId(), dep.getArtifactId(), dep.getClassifier(), dep.getVersion()
-            );
-        }
-        return ret;
+        return lclf.equals(rclf)
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @yegor256 
```java
return Objects.equals(left.getClassifier(), right.getClassifier()) && Objects.equals(left.getArtifactId(), right.getArtifactId()) && Objects.equals(left.getGroupId(), right.getGroupId())
```

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java (hunk 1 above)
Comment (yegor256) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @volodya-lombrozo nope, the tests fail with this

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @yegor256 But the method is correct:
```java
public class DepTest {
    @Test
    void test() {
        Assertions.assertTrue(
            eqTo(
                dep(null, "artifact", "group"),
                dep(null, "artifact", "group")
            )
        );
        Assertions.assertTrue(
            eqTo(
                dep(null, null, "group"),
                dep(null, null, "group")
            )
        );
        Assertions.assertTrue(
            eqTo(
                dep(null, null, null),
                dep(null, null, null)
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep(null, "artifact", "group"),
                dep("classifier", "artifact", "group")
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep("classifier", "artifact", "group"),
                dep(null, "artifact", "group")
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep(null, null, "group"),
                dep(null, "artifact", "group")
            )
        );
        Assertions.assertFalse(
            eqTo(
                dep(null, null, "group"),
                dep(null, null, null)
            )
        );
    }

    private Dependency dep(String classifier, String artifact, String group) {
        final Dependency dep = new Dependency();
        dep.setClassifier(classifier);
        dep.setArtifactId(artifact);
        dep.setGroupId(group);
        return dep;
    }

    private boolean eqTo(Dependency left, Dependency right) {
        return Objects.equals(left.getClassifier(), right.getClassifier())
            && Objects.equals(left.getArtifactId(), right.getArtifactId())
            && Objects.equals(left.getGroupId(), right.getGroupId());
    }
}
```

Comment (yegor256) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @volodya-lombrozo try this:

```
Assertions.assertTrue(
            eqTo(
                dep("", "artifact", "group"),
                dep(null, "artifact", "group")
            )
        );
```

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @yegor256 
```java
    private boolean eqTo(Dependency left, Dependency right) {
        return Objects.equals(
            Optional.ofNullable(left.getClassifier()).orElse(""),
            Optional.ofNullable(right.getClassifier()).orElse("")
        ) && Objects.equals(left.getArtifactId(), right.getArtifactId())
            && Objects.equals(left.getGroupId(), right.getGroupId());
    }

```

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @yegor256 or
```java
    private boolean eqTo(Dependency left, Dependency right) {
        return Objects.equals(
            Objects.toString(left.getClassifier(), ""),
            Objects.toString(right.getClassifier(), "")
        ) && Objects.equals(left.getArtifactId(), right.getArtifactId())
            && Objects.equals(left.getGroupId(), right.getGroupId());
    }
```

Comment (yegor256) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @volodya-lombrozo do you think this code is more readable than the code we have now in the branch?

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @yegor256 Yes, I think it is.  Less variables. Less lines

Comment (yegor256) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @volodya-lombrozo line count is not always the indicator of code readability. This boolean condition is complex as it is. Making it even more complex is unnecessary, I believe.

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @yegor256 what about redundant variables then?

Comment (yegor256) [reply eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java:227]: @volodya-lombrozo I fixed the code, it looks more or less OK in the second option you suggested

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**